packages/pluginmanager-extension/src/index.ts

```diff
@@ -0,0 +1,170 @@
+    const refreshPlugins = trans.__('Refresh Plugin List');
+
+    const namespace = 'plugin-manager';
+    const tracker = new WidgetTracker<MainAreaWidget<Plugins>>({
+      namespace: namespace
+    });
+
+    /**
+     * Create a MainAreaWidget for Plugin Manager.
+     */
+    function createWidget(args?: PluginListModel.IConfigurableState) {
+      const model = new PluginListModel({
+        ...args,
+        pluginData: {
+          availablePlugins: app.info.availablePlugins
+        },
+        serverSettings: app.serviceManager.serverSettings,
+        extraLockedPlugins: [
+          PLUGIN_ID,
+          // UI will not proceed beyond splash without `layout` plugin
+          '@jupyterlab/application-extension:layout',
+          // State restoration does not work well without resolver,
+          // can leave user locked out of the plugin manager
+          // (if command palette and menu are disabled too)
+          '@jupyterlab/apputils-extension:resolver'
+        ],
+        translator: translator ?? nullTranslator
+      });
+      const content = new Plugins({
+        model,
+        translator: translator ?? nullTranslator
+      });
+      content.title.label = widgetLabel;
+      content.title.icon = extensionIcon;
+      const main = new MainAreaWidget({ content, reveal: model.ready });
+
+      main.toolbar.addItem(
+        'refresh-plugins',
+        new CommandToolbarButton({
+          id: CommandIDs.refreshPlugins,
+          args: { noLabel: true },
+          commands
+        })
+      );
+
+      return main;
+    }
+
+    // Register commands.
+    commands.addCommand(CommandIDs.open, {
+      label: widgetLabel,
+      execute: args => {
+        const main = createWidget(args);
+        shell.add(main, 'main', { type: 'Plugins' });
+
+        // add to tracker so it can be restored, and update when choices change
+        void tracker.add(main);
+        main.content.model.trackerDataChanged.connect(() => {
+          void tracker.save(main);
+        });
+        return main;
+      }
+    });
+
+    commands.addCommand(CommandIDs.refreshPlugins, {
+      label: args => (args.noLabel ? '' : refreshPlugins),
+      caption: trans.__('Refresh plugins list'),
+      icon: refreshIcon,
+      execute: async () => {
+        return tracker.currentWidget?.content.model
+          .refresh()
+          .catch((reason: Error) => {
+            console.error(
+              `Failed to refresh the available plugins list:\n${reason}`
+            );
+          });
+      }
+    });
+
+    if (palette) {
+      palette.addItem({ command: CommandIDs.open, category });
+    }
+
+    if (menu) {
+      const settingsMenu = menu.settingsMenu;
+      settingsMenu.addItem({ command: CommandIDs.open });
+    }
+
+    if (restorer) {
+      void restorer.restore(tracker, {
+        command: CommandIDs.open,
+        name: _ => 'plugins',
+        args: widget => {
+          const { query, isDisclaimed } = widget.content.model;
```

Comment:
Should `isDisclaimed` be saved as settings rather than a state variable?

Reply:
The intent was to have something which is semi-persisted as a trade-off between requiring users to tick it every time, or only once, therefore I think it should remain a state variable. Of course if we decide to make it a setting later, it will be possible (and it is easier to add a setting than remove one).